tests/cases/boolean/bool_and.test

```diff
@@ -1,4 +1,4 @@
-### SUBSTRAIT_SCALAR_TEST: v1.0
+### SUBSTRAIT_AGGREGATE_TEST: v1.0
```

Comment:
`bool.and` is a scalar function with variadic args. We should fix the tests.

Reply:
This is bool.bool_and:  https://github.com/substrait-io/substrait/blob/main/extensions/functions_boolean.yaml#L118

Reply:
oh I missed the _ part.